contrib/golang/filters/http/source/golang_filter.cc

```diff
@@ -910,6 +910,20 @@ CAPIStatus Filter::getStringValue(int id, GoString* value_str) {
   case EnvoyValue::DownstreamRemoteAddress:
     req_->strValue = state.streamInfo().downstreamAddressProvider().remoteAddress()->asString();
     break;
+  case EnvoyValue::UpstreamHostAddress:
+    if (state.streamInfo().upstreamInfo() && state.streamInfo().upstreamInfo()->upstreamHost()) {
+      req_->strValue = state.streamInfo().upstreamInfo()->upstreamHost()->address()->asString();
+    } else {
+      return CAPIStatus::CAPIValueNotFound;
+    }
+    break;
+  case EnvoyValue::UpstreamClusterName:
+    if (state.streamInfo().upstreamInfo() && state.streamInfo().upstreamInfo()->upstreamHost()) {
+      req_->strValue = state.streamInfo().upstreamInfo()->upstreamHost()->cluster().name();
```

Comment:
Try to get cluster info from stream info directly. It's possible that cluster is selected but no upstream host.

Reply:
done, thanks.